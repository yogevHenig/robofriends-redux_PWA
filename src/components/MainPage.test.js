//import {shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';
import MainPage from './MainPage';

let testRenderer;
let testInstance;

beforeEach( () => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false,
	}
	testRenderer = renderer.create(<MainPage { ...mockProps } />);
	testInstance = testRenderer.root;

})

it('renders MainPage without crashing', () => {
	expect(testRenderer.toJSON()).toMatchSnapshot();
})

it('filters robots. correctly', () => {
const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id:3,
			name:'john@gmail.com',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		isPending: false,
	}
	const testRenderer2 = renderer.create(<MainPage { ...mockProps2 } />);
	const testInstance2 = testRenderer2.root;

	expect(testInstance.instance.filterRobots()).toEqual([])
	expect(testInstance2.instance.filterRobots()).toEqual([{
			id:3,
			name:'john@gmail.com',
			email: 'john@gmail.com'
		}])
})

it('pending to robots from the api', () => {
const mockProps3 = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: true,
	}
	const testRenderer3 = renderer.create(<MainPage { ...mockProps3 } />);
	const testInstance3 = testRenderer3.root;
	expect(testRenderer3.toJSON()).toMatchSnapshot();
})


// yogi test coverage npm test -- --coverage --watchAll=false
