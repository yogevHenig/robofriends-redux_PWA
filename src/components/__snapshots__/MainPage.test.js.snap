// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pending to robots from the api 1`] = `
<div
  className="tc"
>
  <div>
    <h1
      className="f1"
    >
      RoboFriends
    </h1>
    <button
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
  <div
    className="pa2"
  >
    <input
      aria-label="Search"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <h1>
      Loading
    </h1>
  </div>
</div>
`;

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <div>
    <h1
      className="f1"
    >
      RoboFriends
    </h1>
    <button
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
  <div
    className="pa2"
  >
    <input
      aria-label="Search"
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
